src/PAModel/IR/IRStateHelpers.cs

```diff
@@ -196,6 +197,15 @@ private static void SplitIRAndState(ControlInfoJson.Item control, string topPare
                 entropy.OverridablePropertiesEntry.Add(control.Name, OverridablePropVal);
             }
 
+            // Storing pcf controls template data in entropy
+            // Since this could be different for different controls, even if that appear to follow the same template
+            // Eg:Control Instance 1 -> pcftemplate1 -> DynamicControlDefinitionJson1
+            // Control Instance 2 -> pcftemplate1 -> DynamicControlDefinitionJson2
+            if (IsPCFControl(control.Template))
+            {
+                entropy.PCFTemplateEntry.Add(control.Name, control.Template);
```

Comment:
Just to double check - is this template being stored anywhere else today? Will it be stored in both this `PCFTemplateEntry` collection and another?

Reply:
That is a good point. Let me try removing from the other collection (templatestore) and check if no sideeffects.

Reply:
I think you still need template store for components and controls though, right?